rotkehlchen/exchanges/htx.py

```diff
@@ -188,13 +204,135 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
 
         return returned_balances, ''
 
+    def _paginated_query(
+            self,
+            endpoint: Literal[
+                '/v1/query/deposit-withdraw',
+                '/v1/order/matchresults',
+            ],
+            options: dict[str, Any],
+    ) -> list[dict[str, Any]]:
+        """
+        Query endpoints that can return paginated data. At the moment this is used for asset
+        movements. For asset movements we use cursor pagination since for any time range
+        the api will return a valid response and we don't know when to stop querying.
+        The cursor pagination uses the id of the last object in the last
+        query to return the results of the next page.
+        """
+        result = []
+        # copy since we are modifying the dict and could affect other queries using
+        # the same options
+        query_options = options.copy()
+        while True:
+            output = self._query(absolute_path=endpoint, options=query_options)
+            if len(output) == 0:
+                break
+
+            query_options['from'] = output[-1]['id']
+
+            result.extend(output)
+            if len(output) < PAGINATION_LIMIT:
+                break
+
+        return result
+
+    def _query_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
```

Comment:
this function can raise RemoteError and I believe it is not handled... is it somewhere in an upper level of the logic? State it in the docstring